java/arcs/core/data/CapabilitiesNew.kt

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+package arcs.core.data
+
+/** Store capabilities containing a grouping of individual capabilities */
+class CapabilitiesNew(val ranges: List<CapabilityNew.Range> = emptyList()) {
+
+    val persistence: CapabilityNew.Persistence?
+        get() {
+            return ranges.find { it.min is CapabilityNew.Persistence }?.let {
+                require(it.min.isEquivalent(it.max)) { "Cannot get Persistence for a range" }
+                return it.min as CapabilityNew.Persistence
+            }
```

Comment:
You could factor this out into a helper method:

```
    val persistence: CapabilityNew.Persistence?
        get() = getCapability<CapabilityNew.Persistence>()

    private inline fun <reified T : CapabilityNew> getCapability(): T? {
        return ranges.find { it.min is T }?.let {
            require(it.min.isEquivalent(it.max)) { "Cannot get capability for a range" }
            it.min as T
        }
    }
```

Reply:
thank you! this is so much cleaner, i just couldn't figure out a kt way to do it before.